src/storage/heap_file.c

```diff
@@ -22355,9 +22355,17 @@ heap_insert_logical (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONTEXT * context,
    */
   /* make sure we have IX_LOCK on class see [NOTE-1] */
   // todo - this should probably be disabled in case of bulk insert
-  if (lock_object (thread_p, &context->class_oid, oid_Root_class_oid, IX_LOCK, LK_UNCOND_LOCK) != LK_GRANTED)
+  if (context->is_bulk_op)
     {
-      return ER_FAILED;
+      // In case of bulk insert we need to skip the IX lock on class and make sure that we have BU_LOCK acquired.
+      assert (locator_is_BU_locked (context->class_oid));
```

Comment:
replace locator_is_BU_locked with lock_has_lock_on_object.  locator_is_BU_locked was a work-around used to trick system into thinking that loaddb workers have BU locks :). Now that they really do, it shouldn't be needed.